pkg/docker/registry.go

```diff
@@ -104,6 +105,13 @@ func getUpdatedEndpoint(originalEndpoint, image string) string {
 	return originalEndpoint
 }
 
+// Replace original registries with the new endpoint from the registry mirror configuration.
+func getLocalImage(endpoint string, image string) string {
+	replacer := strings.NewReplacer(defaultRegistry, endpoint, packageProdDomain, endpoint, packageDevDomain, endpoint)
```

Comment:
curious, why not just always replace the host, irregardless of which one it is? That way we make this agnostic of the different ecr domains, it just knows that it need to replace the domain with `endpoint`

Reply:
Not sure if there was an implementation detail here. I just added extra endpoints to an already existing functionality. I can try urls.replacehost and see if that would work the same mechanism.

Reply:
Looks like there is an issue with url parsing that causes an issue. I put a comment on the code to indicate the issue.